### ISSUE TITLE
Simplify the security config required for login forms

### ISSUE THREAD

**Issue by javiereguiluz** [symfony/symfony#27602]
**Description**  
Simplify the security config of a login form.

**Example**  
It's common to use this config when setting up a traditional login form:

```yaml
security:
    firewalls:
        main:
            # ...
            form_login:
                check_path:  some_route_name
                login_path:  some_route_name
```

Could we make `check_path` optional and initialize it to the same value as `login_path` by default? If some user needs to customize this default value, just define the `check_path` explicitly as all developers are forced to do now.

**Comment by iltar** [comment]
Isn't `form_login` one of the first variants that exists? Imo this variant is better off being deprecated and be replaced by the guard variant. I believe guard actually has a `form_login` as well, but I think @weaverryan can explain more of that. I believe Guard has more control over this and doesn't enforce a check URL, but I could be wrong.

**Comment by weaverryan** [comment]
I agree with @iltar. I'm more interested in making Guard super easy to setup (because once setup, it's easier to understand & extend). We already have `AbstractFormLoginAuthenticator` that takes care of a lot of work. Then, soon, we'll add a `make:guard:login` command to give you that skeleton (and will also optionally generate the route/controller/template).

I think that at least "mostly" takes care of the problem. CSRF protection is another thing that is handled more-or-less automatically with `form_login`. We can generate the code needed for that with `make:guard:login`, but there may be some feature (e.g. trait) that could also be added to core to make this even easier.

Ultimately, I think the best path forward is to go with Guard, stop recommending the built-in auth listeners, and try to package generators or traits/base classes in Symfony to make the many "flavors" of auth very easy :).

WDYT?

**Comment by javiereguiluz** [comment]
@iltar @weaverryan I'm sure your proposals are the right solution ... but they will also require dedicating some time to implement, finding someone willing to do that, etc.

Using pseudo-code, this is my proposal:

```php
if (isset(config['form_login.login_path']) && !isset(config['form_login.check_path'])) {
    config['form_login.check_path'] = config['form_login.login_path'];
}
```

I think it may be useful for users and easy enough to implement. What do you think?

**Comment by weaverryan** [comment]
This indeed makes sense to me - so +1 :)

I’m also already working on the generator part of the Guard stuff :)

**Comment by iltar** [comment]
@javiereguiluz that does seem like an easy fix, so why not!

**Comment by javiereguiluz** [comment]
We tried to fix this in #27617 but we can't do that simply and it'll introduce BC breaks. Let's close this as "won't fix". Setting this `check_path` option is not that complicated after all 😝
